src/ray/core_worker/task_manager.cc

```diff
@@ -1030,6 +1034,18 @@ bool TaskManager::RetryTaskIfPossible(const TaskID &task_id,
   }
 }
 
+void TaskManager::RetryTask(TaskEntry *task_entry,
+                            bool object_recovery,
+                            uint32_t delay_ms) {
+  RAY_CHECK(task_entry != nullptr);
+  SetTaskStatus(*task_entry,
+                rpc::TaskStatus::PENDING_ARGS_AVAIL,
+                /* state_update */ std::nullopt,
+                /* include_task_info */ true,
+                task_entry->spec.AttemptNumber() + 1);
```

Comment:
Can we update task_entry attempt number first before calling `SetTaskStatus` so we don't need to manually pass in a different attempt number.

Reply:
We can, but the semantics of `retry_task_callback` are a bit odd: it expects you to update the attempt number before calling it. It may be fine if `RetryTask` is the only place to call the callback.

Do you have any preference?

Reply:
Discussed offline, more cleanups can be done in folloow-ups.